# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../spqmail/0003-qmail-smtpd-ehlo-tls-auth-started.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From a3e48b140cc5b61fb3d1261e8dec8500b7a7c10b Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Fri, 16 Nov 2012 18:15:21 +0100
Subject: [PATCH] qmail-smtpd: fixed smtp_ehlo() to provide correct sasl
 methods for TLS mode if TLS is already started

---
 qmail-smtpd.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/qmail-smtpd.c b/qmail-smtpd.c
index a21fb83..3a1f410 100644
--- a/qmail-smtpd.c
+++ b/qmail-smtpd.c
@@ -390,6 +390,8 @@ void smtp_ehlo(char *arg)
     if (ctl_auth_oldauth) {
       doserv(); out("250-AUTH=LOGIN"); doeol(); /* for old progs */
     }
+    /* provide sasl methods for TLS mode if TLS is already started */ 
+    if (tls_started) ctl_auth_current = &ctl_auth_secure;
     doserv(); out("250-AUTH "); dosasl(); doeol(); /* rfc */
   }
   out_err("250-ENHANCEDSTATUSCODES");
-- 
1.7.10.2

